LookAndFeels/foleys_Skeuomorphic.cpp

```diff
@@ -0,0 +1,240 @@
+    are permitted provided that the following conditions are met:
+    1. Redistributions of source code must retain the above copyright notice, this
+       list of conditions and the following disclaimer.
+    2. Redistributions in binary form must reproduce the above copyright notice,
+       this list of conditions and the following disclaimer in the documentation
+       and/or other materials provided with the distribution.
+    3. Neither the name of the copyright holder nor the names of its contributors
+       may be used to endorse or promote products derived from this software without
+       specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+    IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+    INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+    DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+    LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+    OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+    OF THE POSSIBILITY OF SUCH DAMAGE.
+ ==============================================================================
+ */
+
+namespace foleys
+{
+using namespace juce;
+
+static inline void clipToEllipse (Graphics& g, Rectangle<float>& bounds)
+{
+    Path p;
+    p.addEllipse (bounds);
+    g.reduceClipRegion (p);
+}
+
+static inline ColourGradient makeGradient (Point<float> point1,
+                                           Point<float> point2,
+                                           bool isRadial,
+                                           std::initializer_list<std::pair<double, Colour>> list)
+{
+    ColourGradient gr {};
+    gr.isRadial = isRadial;
+    gr.point1 = point1;
+    gr.point2 = point2;
+
+    for (auto pair : list)
+        gr.addColour (pair.first, pair.second);
+
+    return gr;
+}
+
+static inline void fillEllipse (Graphics& g, const Rectangle<float>& bounds, Colour colour)
+{
+    g.setColour (colour);
+    g.fillEllipse (bounds);
+}
+
+static inline void fillEllipse (Graphics& g, const Rectangle<float>& bounds, const ColourGradient& gradient)
+{
+    g.setGradientFill (gradient);
+    g.fillEllipse (bounds);
+}
+
+const Skeuomorphic::knobImages& Skeuomorphic::getKnobImages (int diameter)
+{
+    auto search = knobsBgs.find (diameter);
+    if (search != knobsBgs.end()) return search->second;
+
+    float radius = diameter * 0.5f;
+    bool isTiny  = diameter < 20;
+    bool isSmall = diameter < 60;
+
+    Point<float>     centre { radius, radius };
+    Rectangle<float> bounds { 0, 0, float (diameter), float (diameter) };
```

Comment:
Either all floats or none ;-)
0.0f...

Reply:
Makes sense, I don't even know how that compiled since I'm using an initializer list.